File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/UnsupportedOneOf.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.base.destination.typing_deduping;

import java.util.List;

/**
 * Represents a {oneOf: [...]} schema.
 * <p>
 * This is purely a legacy type that we should eventually delete. See also {@link Union}.
 */
public record UnsupportedOneOf(List<AirbyteType> options) implements AirbyteType {

}

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/UnsupportedOneOf.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.base.destination.typing_deduping

/**
 * Represents a {oneOf: [...]} schema.
 *
 * This is purely a legacy type that we should eventually delete. See also [Union].
 */
data class UnsupportedOneOf(val options: List<AirbyteType>) : AirbyteType {
    override val typeName: String = TYPE

    companion object {
        const val TYPE: String = "UNSUPPORTED_ONE_OF"
    }
}
